c_glib/arrow-glib/record-batch.cpp

```diff
@@ -191,6 +191,27 @@ garrow_record_batch_equal(GArrowRecordBatch *record_batch,
   return arrow_record_batch->Equals(*arrow_other_record_batch);
 }
 
+/**
+ * garrow_record_batch_equal_metadata:
+ * @record_batch: A #GArrowRecordBatch.
+ * @other_record_batch: A #GArrowRecordBatch to be compared.
+ * @check_metadata: Whether to compare metadata.
+ *
+ * Returns: %TRUE if both of them have the same data, %FALSE
+ *   otherwise.
+ *
+ * Since: 0.17.0
+ */
+gboolean
+garrow_record_batch_equal_metadata(GArrowRecordBatch *record_batch,
+                                   GArrowRecordBatch *other_record_batch,
+                                   gboolean check_metadata)
```

Comment:
Could you also add the following prototype declaration to `record-batch.h`?

```c
GARROW_AVAILABLE_IN_0_17
gboolean
garrow_record_batch_equal_metadata(GArrowRecordBatch *record_batch,
                                   GArrowRecordBatch *other_record_batch,
                                   gboolean check_metadata);
```

Reply:
done